### PR TITLE
Bug Fixes, Added RPS and Config

### PR DESCRIPTION
**Pull request by TheEmperor342**
## Changlog
- \+ Added Rock-Paper-Scissors
- \+ Added Configuration Cog
- \+ Updated Help
- \+ Fixed all link related bugs in `f` command
- \+ There is now has a chance of 30% to earn XP
- \- Removed `memes` command

btw the `Configuration/Leveling.json` is intentional here

### CODE REVIEW

**Comment by itsmebirdie** [comment]
Acknowledged.

File: Cogs/Fun.py [hunk 1]
```diff
@@ -311,16 +610,18 @@ async def coin(self, ctx: commands.Context) -> None:
     @commands.command()
     async def f(self, ctx: commands.Context, *, reason: str = None) -> None:
         if reason is not None:
-            if reason.__contains__("https://"):
+            if any(k in reason for k in ["https://", "http://", "discord.com", "discord.gg"]):
```
Comment (itsmebirdie) [inline Cogs/Fun.py:629]: I would recommend returning an error like:
`There's a Discord invite link in the reason` when having `discord.gg` or `discord.com/invite` in the reason

File: Cogs/Fun.py [hunk 2]
```diff
@@ -340,23 +645,17 @@ async def eightBall(self, ctx: commands.Context, *, question) -> None:
...
-                res = r.json()
-                embed_ = discord.Embed(title=res['title'], color=discord.Color.blue())
-                embed_.set_image(url = res["image"])
-                embed_.set_author(name = ctx.author,icon_url = ctx.author.avatar_url)
-                await ctx.send(embed = embed_)
+        while True:
+            r = requests.get("https://memes.blademaker.tv/api?lang=en")
+            res = r.json()
+            if res["nsfw"]:
+                continue
```
Comment (itsmebirdie) [inline Cogs/Fun.py:668]: Please may I ask an explanation of what it does? 😄

**Review by itsmebirdie — COMMENTED** [review]
Review done.
Awaiting testing.

In the meantime please take a look at the comments 😄

File: Dockerfile [hunk 3]
```diff
@@ -3,6 +3,6 @@ LABEL maintainer="0xbirdie@gmail.com"
 
 ADD . .
 RUN apt update && apt install -y ffmpeg
-RUN pip install discord.py[voice] requests dpymenus youtube_dl disrank numpy pandas
+RUN pip install discord.py[voice] requests dpymenus youtube_dl disrank numpy pandas discord-components
```
Comment (itsmebirdie) [inline Dockerfile]: Hey, this was already committed I guess..

**Review by itsmebirdie — CHANGES_REQUESTED** [review]
Testing failed.

image
image

File: Dockerfile (hunk 3 above)
Comment (TheEmperor342) [reply Dockerfile]: I haven't even touched this file though

File: Cogs/Fun.py (hunk 2 above)
Comment (TheEmperor342) [reply Cogs/Fun.py:668]: In the json from the request of the api there is a key called `nsfw` which is `True` if the image is nsfw

if it is true,
the while True loop is continued 

if it is not the image is sent as an embed and the loop is broken

**Review by itsmebirdie — CHANGES_REQUESTED** [review]
The bot is randomly throwing errors
image
image

**Comment by itsmebirdie** [comment]
Acknowledged commits.

**Review by itsmebirdie — COMMENTED** [review]
Code review done.
Awaiting testing.

**Review by itsmebirdie — CHANGES_REQUESTED** [review]
Testing failed.
Fix requested for:
image


when specifying a help category it shows the default help menu instead of the specific category help.

**Comment by itsmebirdie** [comment]
Also the bot is being rate limited when adding all the reactions in `ttt`
It is better to use the new buttons instead of reactions

**Comment by TheEmperor342** [comment]
> Testing failed.
> Fix requested for:
> image
> 
> when specifying a help category it shows the default help menu instead of the specific category help.

That is intentional. Choose what you want from the select

**Comment by TheEmperor342** [comment]
> Testing failed.
> Fix requested for:
> image
> 
> when specifying a help category it shows the default help menu instead of the specific category help.

Okay

**Review by itsmebirdie — APPROVED** [review]
Approved for merge.
Good Job!

**Pull request merged**